pyteal/ast/subroutine.py

```diff
@@ -269,15 +292,39 @@ def has_return(self):
 SubroutineDeclaration.__module__ = "pyteal"
 
 
+@dataclass
+class _OutputKwArgInfo:
+    name: str
+    abi_type: abi.TypeSpec
+
+    @staticmethod
+    def from_dict(kwarg_info: dict[str, abi.TypeSpec]) -> Optional["_OutputKwArgInfo"]:
+        if kwarg_info is None or len(kwarg_info) == 0:
+            return None
+        elif len(kwarg_info) == 1:
+            key = list(kwarg_info.keys())[0]
+            return _OutputKwArgInfo(key, kwarg_info[key])
+        else:
+            raise TealInputError(
+                f"illegal conversion kwarg_info length {len(kwarg_info)}."
+            )
+
+
+_OutputKwArgInfo.__module__ = "pyteal"
+
+
 class SubroutineCall(Expr):
     def __init__(
         self,
         subroutine: SubroutineDefinition,
-        args: List[Union[Expr, ScratchVar, abi.BaseType]],
+        args: list[Expr | ScratchVar | abi.BaseType],
+        *,
+        output_kwarg: Optional[_OutputKwArgInfo] = None,
```

Comment:
```suggestion
        output_kwarg: _OutputKwArgInfo = None,
```
I believe **mypy** is clever enough to understand that your type is `Optional` when defaulting to `None` (but don't trust me on that)

Reply:
fixed